homeassistant/components/myuplink/sensor.py

```diff
@@ -0,0 +1,161 @@
+"""Sensor for myUplink."""
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from myuplink.models import DevicePoint
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from . import MyUplinkDataCoordinator
+from .const import DOMAIN
+from .coordinator import CoordinatorData
+from .entity import MyUplinkEntity
+
+
+@dataclass(kw_only=True)
+class MyUplinkDeviceSensorEntityDescription(SensorEntityDescription):
+    """Describes MyUplink device sensor entity."""
+
+    value_fn: Callable[[CoordinatorData, str], StateType]
+
+
+DEVICE_SENSORS: tuple[MyUplinkDeviceSensorEntityDescription, ...] = (
+    MyUplinkDeviceSensorEntityDescription(
+        key="firmware_current",
+        name="Firmware Current",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data, device_id: data.devices[device_id].firmwareCurrent,
+    ),
+    MyUplinkDeviceSensorEntityDescription(
+        key="firmware_desired",
+        name="Firmware Desired",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data, device_id: data.devices[device_id].firmwareDesired,
+    ),
+    MyUplinkDeviceSensorEntityDescription(
+        key="connection_state",
+        name="Connection state",
```

Comment:
What does the connection state value represent and what values can it take?

Reply:
From the myuplink swagger data it looks like it's a two item enum: Connected or Disconnected.

In that case this should be a binary sensor with device class `BinarySensorDeviceClass.CONNECTIVITY` instead.

https://developers.home-assistant.io/docs/core/entity/binary-sensor#available-device-classes